### ISSUE TITLE
Cannot use `Required` and `NotRequired` with Python versions < 3.10

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#2931]
The following code works for Python 3.7-3.9:

```python
from __future__ import annotations  # for Python 3.7-3.9
from typing import TypedDict

class Dog(TypedDict):
    name: str
    owner: NotRequired[str|None]
```

However Pyright wrongly reports an error for the last line (when e.g. invoked with `--pythonversion 3.9`):

```
error: "NotRequired" is not defined (reportUndefinedVariable)
```

The code example is directly out of [PEP 655](https://www.python.org/dev/peps/pep-0655/#how-to-teach-this) (support for which was introduced in Pyright with 7ed245b1845173090c6404e49912e8cbfb3417c8).

**Comment by not-my-profile** [comment]
Ah my bad ... the solution is to use [`typing_extensions`](https://github.com/python/typing/blob/master/typing_extensions/README.rst#typing-extensions).
